sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -379,17 +383,46 @@ private object CosmosReadConfig {
   private val MaxItemCount = CosmosConfigEntry[Int](
     key = CosmosConfigNames.ReadMaxItemCount,
     mandatory = false,
-    defaultValue = Some(DefaultMaxItemCount),
+    defaultValue = None,
     parseFromStringFunction = queryText => queryText.toInt,
     helpMessage = "The maximum number of documents returned in a single request. The default is 1000.")
 
+  private val PrefetchBufferSize = CosmosConfigEntry[Int](
+    key = CosmosConfigNames.ReadPrefetchBufferSize,
+    mandatory = false,
+    defaultValue = None,
+    parseFromStringFunction = queryText => queryText.toInt,
+    helpMessage = "The prefetch buffer size - this limits the number of pages (max. 5 MB per page) that are " +
+      s"prefetched from the Cosmos DB Service. The default is `1` if the '${CosmosConfigNames.ReadMaxItemCount}' " +
+      "parameter is specified and larger than `1000` and `8` otherwise. If the provided value is not `1` internally " +
```

Comment:
if the maxitemCount > 1000, then no matter what the value the customer set, we are going to use 1 always

Reply:
No - the getOrElse in Line 414 means we choose whatever provided by the customer - the match...- algorithm just described what default value to use if the cx didn't provide any value.

Reply:
Makes sense?

Reply:
yes, misread the line